src/iron.rs

```diff
@@ -58,7 +75,13 @@ impl<H: Handler> Iron<H> {
         let SocketAddr { ip, port } = sock_addr.clone();
         self.addr = Some(sock_addr);
 
-        Ok(try!(Server::http(ip, port).listen_threads(self, threads)))
+        let server = match self.security {
+            Security::Insecure => Server::http(ip, port),
+            Security::Secure { ref certificate, ref key } =>
+                Server::https(ip, port, certificate.clone(), key.clone())
```

Comment:
Do you mean using two enums, one including the certificate and key and one not ("Protocol" and "ProtocolOptions" or some such)? We need to store the current protocol for use in handle(), so if we only use one enum type we end up storing the cert/key paths too.